AEPServices/Sources/storage/NamedCollectionDataStore.swift

```diff
@@ -178,8 +178,9 @@ public class NamedCollectionDataStore {
         }
 
         let encoder = JSONEncoder()
-        let encodedValue = try? encoder.encode(value)
-        set(key: key, value: encodedValue)
+        guard let encodedValue = try? encoder.encode(value) else {return}
+        let encodedString = String(data: encodedValue, encoding: .utf8)
+        set(key: key, value: encodedString)
```

Comment:
in previous code a failure to encode resulted in a nil value set (removal) -- is the change (failing to set and bailing due to the new guard statement) intentional, or was the previous behavior preferred/depended on?

Reply:
This is a good point. Not too sure if we were depending on this to be nil in failure scenarios. What do you think @praveek? Should we remove the guard and set nil when encoding fails?

Reply:
I too prefer simply returning instead of setting nil value. However, I would like to maintain the same behavior as before since I'm uncertain whether other extensions depend on the previous behavior and want to keep all the changes for this migration only within Core.